### PR TITLE
♻️ Refactor Updating

### LINKED ISSUE #1367: Add progressbar when downloading updates
#### Describe your issue

When prompted to do an update there is no way to tell when its finshed!

#### How can others reproduce the issue?

Update bpmn-studio

#### Possible solution

Add an indicator that there is some progress while downloading updates.

### PR DESCRIPTION
**Pull request by SteffenKn**
**Changes:**

1. Refactor Updating
2. Add Update Dropdown to Statusbar
3. Make Download Cancelable
4. Add Progressbar

**Issues:**

Closes #1367

PR: #1368 

## How can others test the changes?

- Download [this](https://ci.process-engine.io/blue/organizations/jenkins/process-engine_node-lts%2Fbpmn-studio/detail/feature%2Fadd_progress_to_update/41/artifacts) version
- Run the BPMN-Studio
- Update the BPMN-Studio
- Cancel the update (via the x-Button)
- Restart the update
- Install the update

## PR-Checklist

Please check the boxes in this list after submitting your PR:

- [x] You can merge this PR **right now** (if not, please prefix the title with "WIP: ")
- [x] I've tested **all** changes included in this PR.
- [x] I've also reviewed this PR myself before submitting (e.g. for scrambled letters, typos, etc.).
- [x] I've rebased the `develop` branch with my branch before finishing this PR.
- [x] I've **summarized all changes** in a list above.
- [x] I've mentioned all **PRs, which relate to this one**.
- [x] I've prefixed my Pull Request title is according to [gitmoji guide](https://gitmoji.carloscuesta.me/).

### CODE REVIEW

File: Jenkinsfile [hunk 1]
```diff
@@ -34,7 +34,7 @@ pipeline {
           branch = env.BRANCH_NAME;
           branch_is_master = branch == 'master';
           branch_is_develop = branch == 'develop';
-          branch_is_release = branch.startsWith('release/');
+          branch_is_release = branch.startsWith('release/') || true;
```
Comment (ElRaptorus) [inline Jenkinsfile]: What's this for?

File: package.json [hunk 2]
```diff
@@ -1,7 +1,7 @@
 {
   "name": "bpmn-studio",
   "description": "An Aurelia application for designing BPMN diagrams, which can also be connected to a process engine to execute these diagrams.",
-  "version": "5.1.2",
+  "version": "5.0.0",
```
Comment (ElRaptorus) [inline package.json]: Rebase error, I'm guessing?

File: src/modules/status-bar/status-bar.ts [hunk 3]
```diff
@@ -22,18 +31,54 @@ export class StatusBar {
...
+
+    const applicationRunsInElectron: boolean = (window as any).nodeRequire !== undefined;
+    if (applicationRunsInElectron) {
+      this._ipcRenderer = (window as any).nodeRequire('electron').ipcRenderer;
+
+      this._ipcRenderer.on('update_error', () => {
+        notificationService.showNotification(NotificationType.INFO, 'Update Error!');
+      });
+
+      this._ipcRenderer.on('update_available', (_: Event, version: string) => {
```
Comment (ElRaptorus) [inline src/modules/status-bar/status-bar.ts]:
```suggestion
      this._ipcRenderer.on('update_available', (event: Event, version: string) => {
```

File: package.json (hunk 2 above)
Comment (SteffenKn) [reply package.json]: This is needed for testing the updater.
But this should be removed before this gets merged 👍

File: Jenkinsfile (hunk 1 above)
Comment (SteffenKn) [reply Jenkinsfile]: This was also for testing.
I forgot to remove the testing stuff.

Comment (SteffenKn) [reply Jenkinsfile]:
```suggestion
          branch_is_release = branch.startsWith('release/');
```

File: package.json (hunk 2 above)
Comment (ElRaptorus) [reply package.json]: Ah, I see. Makes sense. :+1:

File: src/modules/status-bar/status-bar.html [hunk 4]
```diff
@@ -5,7 +5,42 @@
...
   <div class="status-bar status-bar--system-macos" id="statusBarContainer">
     <div class="status-bar__left-bar" id="statusBarLeft">
-
+      <a if.bind="updateAvailable" class="update-button" data-toggle="dropdown" aria-haspopup="true" aria-expanded="false">
+        <img class="update-button-icon" src="src/resources/images/icon.png" title.bind="isDownloading ? updateProgressData.percent.toString() + '%' : 'Update available.'">
+      </a>
+      <div class="dropdown-menu update-dropdown" ref="updateDropdown">
+        <template if.bind="!isDownloading && !updateDownloadFinished">
+          <div class="update-dropdown-title-container">
+              <span class="update-dropdown-title">Version ${updateVersion} available</span>
```
Comment (alexanderkasten) [inline src/modules/status-bar/status-bar.html]:
```suggestion
            <span class="update-dropdown-title">Version ${updateVersion} available</span>
```

Comment (SteffenKn) [inline src/modules/status-bar/status-bar.html]:
```suggestion
            <span class="update-dropdown-title">Version ${updateVersion} available</span>
```

File: src/modules/status-bar/status-bar.ts [hunk 5]
```diff
@@ -22,18 +31,54 @@ export class StatusBar {
...
+    if (applicationRunsInElectron) {
+      this._ipcRenderer = (window as any).nodeRequire('electron').ipcRenderer;
+
+      this._ipcRenderer.on('update_error', () => {
+        notificationService.showNotification(NotificationType.INFO, 'Update Error!');
+      });
+
+      this._ipcRenderer.on('update_available', (event: Event, version: string) => {
+        this.updateAvailable = true;
+
```
Comment (SteffenKn) [inline src/modules/status-bar/status-bar.ts]:
```suggestion
```

File: src/modules/status-bar/status-bar.ts [hunk 6]
```diff
@@ -22,18 +31,54 @@ export class StatusBar {
...
+        this.updateVersion = version;
+
+        const message: string = `A new update is available.\nPlease click on the BPMN-Studio icon in the statusbar to start the download.`;
+
+        this._notificationService.showNonDisappearingNotification(NotificationType.INFO, message);
+      });
+
+      this._ipcRenderer.on('update_download_progress', (event: Event, updateProgressData: any) => {
```
Comment (alexanderkasten) [inline src/modules/status-bar/status-bar.ts]:
```suggestion
      this._ipcRenderer.on('update_download_progress', (event: Event, updateProgressData: UpdateProgressData) => {
```

**Pull request merged**